src/components/EventMap.js

```diff
@@ -61,50 +65,60 @@ export default function(store){
     },
     methods: {
       plotEvents() {
-        // wipe out existing plotted events -- probably this can
-        // be done more efficiently
-        if (this.eventsLayer) {
-          this.eventsLayer.clearLayers();
-        }
-
-        this.eventsLayer = L.layerGroup().addTo(this.mapRef);
+        const eventsSource = this.mapRef.getSource("events");
 
-        this.filteredEvents.forEach((event) => {
-          L.circleMarker([event.lat, event.lng], {
-            radius: 5,
-            color: 'white',
-            fillColor: '#ef3030',
-            opacity: 0.8,
-            fillOpacity: 0.7,
-            weight: 2
-          }).addTo(this.eventsLayer);
-        });
+        if (eventsSource) {
+          const eventsAsGeoJson = this.filteredEvents.map((event) => geoJsonHelpers.point([event.lng, event.lat]));
+          eventsSource.setData(geoJsonHelpers.featureCollection(eventsAsGeoJson));
+        }
       },
 
       setMapPositionBasedOnZip() {
-        if (!this.filters.zipcode || !this.zipcodes[this.filters.zipcode]) {
-
-          this.setInitialMapPosition();
-          return;
-        }
+        let latLng = this.zipcodes[this.filters.zipcode];
+        if (latLng) latLng = [latLng[1], latLng[0]];
+        const zoom = (latLng) ? 8 : this.initialZoom;
 
-        const latLng = this.zipcodes[this.filters.zipcode];
-        const zoom = 8;
+        this.mapRef.flyTo({
+          center: latLng || this.initialCoordinates,
+          zoom: zoom
+        });
+      },
 
-        this.mapRef.setView(latLng, zoom);
+      addCustomIcon(icon, name) {
+        const img = new Image();
+        img.src = icon;
+        img.onload = function() {
+          this.mapRef.addImage(name, img);
+        }.bind(this);
       },
 
-      setInitialMapPosition() {
-        const centerOfUS = [37.8, -96.9];
-        const zoom = 4;
+      createEmptyEventsDataSource() {
+        this.mapRef.addSource("events", {
+          "type": "geojson",
+          "data": geoJsonHelpers.featureCollection([]),
+          "cluster": true,
+          "clusterMaxZoom": 8
+        });
+      },
 
-        this.mapRef.setView(centerOfUS, zoom);
+      mapMounted() {
+        this.createEmptyEventsDataSource();
+        this.addCustomIcon(mapMarker, "custom-marker");
+        mapStyles.forEach((style) => this.mapRef.addLayer(style));
+        this.plotEvents();
       }
     },
 
     mounted() {
-      this.mapRef = L.mapbox.map('map', 'mapbox.streets');
-      this.setInitialMapPosition();
+      this.mapRef = new mapboxgl.Map({
+        container: 'map',
+        style: 'mapbox://styles/mapbox/bright-v9',
+        center: this.initialCoordinates,
+        zoom: this.initialZoom
+      });
+
+      this.mapRef.addControl(new mapboxgl.NavigationControl());
+      this.mapRef.on("load", this.mapMounted.bind(this));
```

Comment:
More bind nits -- I think vue has already bound this method, should be ok to drop `.bind(this)`